lib/segment/src/types.rs

```diff
@@ -341,114 +398,26 @@ pub enum PayloadSchemaType {
-        match self {
-            PayloadVariant::Value(x) => vec![x.clone()],
-            PayloadVariant::List(vec) => vec.clone(),
-        }
-    }
-}
-
-/// Structure for converting user-provided payload into internal structure representation
-///
-/// Used to allow user provide payload in more human-friendly format,
-/// and do not force explicit brackets, included constructions, e.t.c.
-///
-/// Example:
-///
-/// ```json
-/// {..., "payload": {"city": "Berlin"}, ... }
-/// ```
-///
-/// Should be captured by `KeywordShortcut`
-///
-#[derive(Debug, Deserialize, Serialize, JsonSchema, PartialEq, Clone)]
-#[serde(rename_all = "snake_case")]
-#[serde(untagged)]
-pub enum PayloadInterface {
-    KeywordShortcut(PayloadVariant<String>),
-    IntShortcut(PayloadVariant<i64>),
-    FloatShortcut(PayloadVariant<f64>),
-    Payload(PayloadInterfaceStrict),
-}
-
-/// Fallback for `PayloadInterface` which is used if user explicitly specifies type of payload
-///
-/// Example:
-///
-/// ```json
-/// {..., "payload": {"city": { "type": "keyword", "value": "Berlin" }}, ... }
-/// ```
-///
-/// Should be captured by `Keyword(PayloadVariant<String>)`
-#[derive(Debug, Deserialize, Serialize, JsonSchema, PartialEq, Clone)]
-#[serde(rename_all = "snake_case")]
-#[serde(tag = "type", content = "value")]
-pub enum PayloadInterfaceStrict {
-    Keyword(PayloadVariant<String>),
-    Integer(PayloadVariant<i64>),
-    Float(PayloadVariant<f64>),
-    Geo(PayloadVariant<GeoPoint>),
-}
-
-// For tests
-impl From<PayloadInterfaceStrict> for PayloadInterface {
-    fn from(x: PayloadInterfaceStrict) -> Self {
-        PayloadInterface::Payload(x)
-    }
-}
-
-impl From<&PayloadInterfaceStrict> for PayloadType {
-    fn from(interface: &PayloadInterfaceStrict) -> Self {
-        match interface {
-            PayloadInterfaceStrict::Keyword(x) => PayloadType::Keyword(x.to_list()),
-            PayloadInterfaceStrict::Integer(x) => PayloadType::Integer(x.to_list()),
-            PayloadInterfaceStrict::Float(x) => PayloadType::Float(x.to_list()),
-            PayloadInterfaceStrict::Geo(x) => PayloadType::Geo(x.to_list()),
-        }
-    }
-}
-
-impl From<&PayloadInterface> for PayloadType {
-    fn from(interface: &PayloadInterface) -> Self {
-        match interface {
-            PayloadInterface::Payload(x) => x.into(),
-            PayloadInterface::KeywordShortcut(x) => PayloadType::Keyword(x.to_list()),
-            PayloadInterface::FloatShortcut(x) => PayloadType::Float(x.to_list()),
-            PayloadInterface::IntShortcut(x) => PayloadType::Integer(x.to_list()),
+// TODO(gvelo): implement for all json types.
+pub fn get_schema_type(value: &Value) -> Option<PayloadSchemaType> {
+    match value {
+        Value::Null => None,
+        Value::Bool(_) => None,
+        Value::Number(value) => {
+            if value.as_i64().is_some() {
+                Some(PayloadSchemaType::Integer);
+            }
+            if value.as_i64().is_some() {
```

Comment:
`serde_json` has separate methods for this
```suggestion
            if value.is_i64() {
                Some(PayloadSchemaType::Integer);
            }
            if value.is_f64() {
```